src/aliased_buffer.h

```diff
@@ -0,0 +1,193 @@
+
+#ifndef SRC_ALIASED_BUFFER_H_
+#define SRC_ALIASED_BUFFER_H_
+
+#include "v8.h"
+
+namespace node {
+
+/**
+ * This class encapsulates the technique of having a native buffer mapped to
+ * a JS object. Writes to the native buffer can happen efficiently without
+ * going through JS, and the data is then available to user's via the exposed
+ * JS object.
+ *
+ * While this technique is computationaly efficient, it is effectively a
+ * write to JS program state w/out going through the standard
+ * (monitored) API. Thus any VM capabilities to detect the modification are
+ * circumvented.
+ *
+ * The encapsulation herein provides a placeholder where such writes can be
+ * observed. Any notification APIs will be left as a future exercise.
+ */
+template <class NativeT, class V8T>
+class AliasedBuffer {
+ public:
+  AliasedBuffer(v8::Isolate* isolate, const size_t count) :
+    isolate_(isolate),
+    count_(count),
+    byte_offset_(0),
+    freeBuffer_(true) {
+    const v8::HandleScope handle_scope(this->isolate_);
+
+    const int sizeInBytes = sizeof(NativeT) * count;
+
+    // allocate native buffer
+    this->buffer_ = reinterpret_cast<NativeT*>(
+      calloc(count, sizeof(NativeT)));
```

Comment:
Could this follow the same pattern as `MaybeStackBuffer`... or even make use of that to avoid the `calloc` where possible?

Reply:
I could, but I'm not following why.  Can you help me understand your concerns w/ using calloc?  I was trying to be consistent w/ original buffers that were alloc'd on heap.

Reply:
Consistency would be the main reason but it's certainly not critical. I don't think the performance difference is going to be enough to worry about.

Reply:
I'm now using `node::UncheckedCalloc`.  Is using `MaybeStackBuffer` still something to investigate?